libs/admin/src/lib/state/admin.actions.ts

```diff
@@ -0,0 +1,74 @@
+import { createAction, props } from '@ngrx/store';
+import { User, Workgroup } from '../services/admin.service';
+
+export const findUser = createAction(
+  '[Asset Search] Find user',
+  props<{
+    userId: string;
+  }>()
+);
+
+export const setUser = createAction(
+  '[Asset Search] Set user',
+  props<{
+    user: User;
+  }>()
+);
+
+export const updateUser = createAction(
+  '[Asset Search] Update User',
+  props<{
+    user: User;
+  }>()
+);
+
+export const listUsers = createAction('[Asset Search] List users');
+
+export const setUsers = createAction(
+  '[Asset Search] Set users',
+  props<{
+    users: User[];
+  }>()
+);
+
+export const findWorkgroup = createAction(
+  '[Asset Search] Find workgroup',
+  props<{
+    workgroupId: number;
+  }>()
+);
+
+export const setWorkgroup = createAction(
+  '[Asset Search] Set workgroup',
+  props<{
+    workgroup: Workgroup;
+  }>()
+);
+
+export const resetWorkgroup = createAction('[Asset Search] Reset workgroup');
+
+export const setWorkgroups = createAction(
+  '[Asset Search] Set workgroups',
+  props<{
+    workgroups: Workgroup[];
+  }>()
+);
+
+export const updateWorkgroup = createAction(
+  '[Asset Search] Update Workgroup',
+  props<{
+    workgroupId: number;
+    workgroup: Omit<Workgroup, 'id'>;
```

Comment:
Why do we omit the `id`, but then specify it as an additional field?

Reply:
The WorkgroupDataBoundary does not allow an id field, so we just send the rest of the body. And instead of mapping the workgroup without the id in the effect or the service, it is done in the component. But i can also remove the id inside the service method

Reply:
So the `workgroup` field actually always has the type `WorkgroupData`? Because if so, then we can just use that type instead of `Omit<...>`.

Reply:
Agreed. This to me again raises the question where to define all of these types. Currenlty, we have them in the model in the backend and the service in the frontend, which is suboptimal. Where should we move them so we dont dublicate?

Reply:
In #159, I already moved all the types into the new `libs/shared2` library. Since we need to merge the two branches anyways later on this week, you could maybe add a TODO that the type needs to be changed to `WorkgroupData`?

Reply:
Will move the types to shared2 after everything is merged to avoid conflicts